llama_index/llms/rungpt.py

```diff
@@ -0,0 +1,228 @@
+from typing import Any, Optional, Sequence, Dict
+from pydantic import Field
+import json
+import requests
+import sseclient
+from llama_index.llms.base import (
+    LLM,
+    LLMMetadata,
+    ChatMessage,
+    ChatResponse,
+    ChatResponseGen,
+    MessageRole,
+    CompletionResponse,
+    ChatResponseAsyncGen,
+    CompletionResponseGen,
+    CompletionResponseAsyncGen,
+    llm_completion_callback,
+    llm_chat_callback,
+)
+from llama_index.constants import DEFAULT_CONTEXT_WINDOW, DEFAULT_NUM_OUTPUTS
+from llama_index.callbacks import CallbackManager
+
+
+class RunGptLLM(LLM):
+    """The opengpt of Jina AI models."""
+
+    model: Optional[str] = Field(description="The rungpt model to use.")
+    endpoint: str = Field(description="The endpoint of serving address.")
+    temperature: float = Field(description="The temperature to use for sampling.")
+    max_tokens: Optional[int] = Field(description="Max tokens model generates.")
+    context_window: int = Field(
+        description="The maximum number of context tokens for the model."
+    )
+    additional_kwargs: Dict[str, Any] = Field(
+        default_factory=dict, description="Additonal kwargs for the Replicate API."
+    )
+    base_url: str = Field(
+        description="The address of your target model served by rungpt."
+    )
+
+    def __init__(
+        self,
+        model: Optional[str] = "rungpt",
+        endpoint: Optional[str] = "0.0.0.0:51002",
+        temperature: float = 0.75,
+        max_tokens: Optional[int] = 256,
+        context_window: int = DEFAULT_CONTEXT_WINDOW,
+        additional_kwargs: Optional[Dict[str, Any]] = None,
+        callback_manager: Optional[CallbackManager] = None,
+    ):
+        super().__init__(
+            model=model,
+            endpoint=endpoint,
+            temperature=temperature,
+            max_tokens=max_tokens,
+            context_window=context_window,
+            additional_kwargs=additional_kwargs or {},
+            callback_manager=callback_manager or CallbackManager([]),
+            base_url="http://" + endpoint,
```

Comment:
nit: might be more robust to check if the endpoint starts with `http` before adding it?